tests/ui/consts/required-consts/collect-in-dead-forget.rs

```diff
@@ -1,12 +1,12 @@
 //@revisions: noopt opt
-//@build-pass
+//@build-fail
 //@[noopt] compile-flags: -Copt-level=0
 //@[opt] compile-flags: -O
-//! This passes without optimizations, so it can (and should) also pass with optimizations.
```

Comment:
The intent was for it to pass with and without optimizations. Making this fail seems like it would be pretty bad for perf as we will always monomorphize all destructors even when we know for sure they won't be called! I think it is quite important to gather "mentioned" items after drop elaboration, otherwise there's no way we are going to get decent performance here.